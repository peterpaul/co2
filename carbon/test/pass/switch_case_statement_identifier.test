/*
 * carbon Compiler and programming language for libco2
 *
 * Copyright (C) 2011 Peterpaul Taekele Klein Haneveld
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */


["stdio.h"] int printf(char * format, ...);
["include.h"] int _ENUM_1, _ENUM_2;

int main() {
	int i;
	for (i = 0; i < 10; i +=1) {
		switch (i) {
		case _ENUM_1:
		case _ENUM_2:
			printf ("Number: %d!\n", i);
			break;
		default:
			printf ("Number: %d\n", i);
			break;
		}
	}
	return 0;
}
