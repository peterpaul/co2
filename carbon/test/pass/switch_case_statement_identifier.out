Number: 0!
Number: 1!
Number: 2
Number: 3
Number: 4
Number: 5
Number: 6
Number: 7
Number: 8
Number: 9
